### PR TITLE
Open bigger set of configured projects when opening composite project for operations that operate over multiple projects like rename

### PR DESCRIPTION
**Pull request by sheetalkamat**
#28261 on top of #32028

- [x] Add tests
- [x] Bigger project opening set means that much longer time to do the operation. So we need some optimizers for this. There are multiple options here and one or many can be chosen.
   - Need some sort of indicator if the operation is exposing something that's outside scope of the project. (eg renaming local function in checker shouldn't need to open this so that we are delaying this as much as possible) 
   - May be send more locations as event (eg in find all references if not rename can handle that)
- [x] We need to have way to unload projects (with some sort of priority based) when the operation might cause too many projects resulting in it running out of memory. This could probably be different PR but then we need some sort of flag to opt out of opening that bigger set so that people have a way to work with the set.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@typescript-bot pack this

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -795,6 +795,12 @@ namespace ts {
             category: Diagnostics.Advanced_Options,
             description: Diagnostics.Disable_use_of_source_files_instead_of_declaration_files_from_referenced_projects
         },
+        {
+            name: "disableSearchSolution",
```
Comment (RyanCavanaugh) [inline src/compiler/commandLineParser.ts]: Name -> `disableSolutionSearching`

File: src/compiler/commandLineParser.ts [hunk 2]
```diff
@@ -795,6 +795,12 @@ namespace ts {
             category: Diagnostics.Advanced_Options,
             description: Diagnostics.Disable_use_of_source_files_instead_of_declaration_files_from_referenced_projects
         },
+        {
+            name: "disableSearchSolution",
+            type: "boolean",
+            category: Diagnostics.Advanced_Options,
+            description: Diagnostics.Disable_searching_solution_for_this_project
```
Comment (RyanCavanaugh) [inline src/compiler/commandLineParser.ts]: "Disable solution searching for this project"

File: src/server/editorServices.ts [hunk 3]
```diff
@@ -383,10 +383,70 @@ namespace ts.server {
     }
 
     interface OriginalFileInfo { fileName: NormalizedPath; path: Path; }
+    interface AncestorConfigFileInfo {
+        /** config file name */ fileName: string;
+        /** path of open file so we can look at correct root */path: Path;
```
Comment (RyanCavanaugh) [inline src/server/editorServices.ts]: Formatting (declaration on following line)

**Comment by RyanCavanaugh** [comment]
LGTM with some name comments

**Comment by DanielRosenwasser** [comment]
What does this change imply? Are operations like rename more accurate now?

**Comment by sheetalkamat** [comment]
This means in our projects setup if you just opened utilities file from `compiler` and rename some functions in there, it would look for other sibling projects since our solution project references them (even though there is no file from services/server or other such projects open)

**Comment by telamonian** [comment]
I just ran into this problem today. How lovely to see that it's already solved. Thank you for working on this!

I was wondering, what release of typescript will this be a part of? I can't seem to find any relevant info

**Comment by orta** [comment]
It should be in 3.8 beta

**Comment by sunniejai** [comment]
Does this resolve https://github.com/microsoft/TypeScript/issues/30823?

**Comment by OliverJAsh** [comment]
I was linked here from https://github.com/microsoft/vscode/issues/80423. I just tested 3.8 RC and the issue I described in https://github.com/microsoft/vscode/issues/80423 still remains.

**Comment by pokey** [comment]
https://github.com/microsoft/TypeScript/issues/30823#issuecomment-1452294101

**Pull request merged**